ckanext/datajson/package2pod.py

```diff
@@ -460,9 +482,17 @@ def generate_distribution(someValue):
                 if 'mediaType' in resource:
                     if 'accessURL' in resource:
                         resource.pop('accessURL')
-                    resource['downloadURL'] = res_url
+                    resource['downloadURL'] = res_url.replace('URL:', '').strip()
+                    if not datajsonvalidator.check_url_field(True, resource, 'downloadURL', None, {}):
+                        resource['downloadURL'] = urllib.quote(res_url.replace('URL:', '').strip())
+                        if not datajsonvalidator.check_url_field(True, resource, 'downloadURL', None, {}):
+                            resource.pop('downloadURL')
                 else:
-                    resource['accessURL'] = res_url
+                    resource['accessURL'] = res_url.replace('URL:', '').strip()
+                    if not datajsonvalidator.check_url_field(True, resource, 'accessURL', None, {}):
+                        resource['accessURL'] = urllib.quote(res_url.replace('URL:', '').strip())
+                        if not datajsonvalidator.check_url_field(True, resource, 'accessURL', None, {}):
+                            resource.pop('accessURL')
```

Comment:
Actually `pop` here is just removing the URL. If you have a resource that doesn't have an access or download url, then it's invalid (and kind of useless). See [schema](https://resources.data.gov/resources/dcat-us/#distribution-accessURL)
You'll want to skip this resource, it gets added at the `arr +=` line. So we need a variable like `bad_resource` flag (defaults to false), gets set to true when the url check fails, and then check that before adding the resource into the `arr`...